client/src/main/java/io/quarkus/vault/client/common/VaultTracingExecutor.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.vault.client.common;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.smallrye.mutiny.Uni;
+
+public class VaultTracingExecutor implements VaultRequestExecutor {
+
+    private static final Logger log = Logger.getLogger(VaultTracingExecutor.class.getName());
+
+    private final VaultRequestExecutor delegate;
+
+    public VaultTracingExecutor(VaultRequestExecutor delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public <T> Uni<VaultResponse<T>> execute(VaultRequest<T> request) {
+        log.info("Executing: " + request.getOperation() + "\n" + getCurlFormattedRequest(request));
+        return delegate.execute(request)
+                .onItem().invoke((response) -> {
+                    var message = "Response: " + request.getOperation() + "\n" + getHTTPFormattedResponse(response) + "\n";
```

Comment:
should that be `System.getProperty("line.separator")`

Reply:
Sure.